### DISCUSSION TITLE
Statistical expressions in report

### DISCUSSION THREAD

**Discussion by quasiquote1** [sparcians/map#386] [Q&A] [answered]
Hi, I'm trying to write long statistical expressions in the report yaml file and find that I will likely need to reference field_names in my expressions. 

Example:

   (lookups - hits) : misses
    (misses * 1000.0 / total_insts) : misses_pki

Where the second statistical expression refers to the "misses" field name in the first key/value pair.

I saw this statement in the documentation:

Other counters/stats can be referenced relative to the current context in the report def just as simple named counters/stats are referenced in basic report entries.

https://sparcians.github.io/map/report_def_format.html#report_def_expressions

Do you have examples where field_names are used for subsequent statistical expressions?

Thanks!

**Comment by klingaard** [comment, marked as answer]
Ugh, you hit on a bug/feature that isn't in Sparta yet: https://github.com/sparcians/map/issues/311  

I have a branch where I started adding this support, but haven't had time to finish it up.  I'll move it up in priority.

**Comment by ghost** [comment]
Still working on this.  😄

**Reply by klingaard** [reply]
Hey @quasiquote1 , I finally fixed/added support for this!  Please give it a try in map_v2.0.12.  https://github.com/sparcians/map/releases/tag/map_v2.0.12
